### PR TITLE
Support Threat Assessment functionality in MS Graph Security

### PR DESCRIPTION
**Pull request by merit-maita**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-2417

## Description
Add support to Threat Assessment functionality in MS Graph Security

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 1]
```diff
@@ -1623,7 +1624,409 @@ script:
...
+    - description: expected assessment.
+      isArray: false
+      name: expected_assessment
+      defaultValue: all
+      auto: PREDEFINED
+      predefined:
+      - 'block'
+      - 'unblock'
+      required: true
+    - description: category.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:1797]:
```suggestion
    - description: The category of the threat.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 2]
```diff
@@ -1623,7 +1624,409 @@ script:
...
+      required: true
+    - description: Base 64 encoded text file.
+      isArray: false
+      name: content_data
+      required: false
+    - description: Entry ID.
+      isArray: false
+      name: entry_id
+      required: false
+    - description: whether to hide the polling result (automatically filled by polling).
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:1818]:
```suggestion
    - description: Whether to hide the polling result (automatically filled by polling).
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 3]
```diff
@@ -1623,7 +1624,409 @@ script:
...
+      name: hide_polling_output
+      deprecated: true
+    - description: request_id (generated in polling command).
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:1821]:
```suggestion
    - description: The ID of the request (generated in the polling command).
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 4]
```diff
@@ -1623,7 +1624,409 @@ script:
...
+      description: 'Created user id.'
+      type: String
+    - contextPath: MSGraphMail.FileAssessment.CreatedUsername
+      description: 'Created user name.'
+      type: String
+    - contextPath: MSGraphMail.FileAssessment.ResultType
+      description: 'Result Type.'
+      type: String
+    - contextPath: MSGraphMail.FileAssessment.ResultMessage
+      description: 'Result message.'
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:1863]:
```suggestion
      description: The mail assessment ID.
      type: String
    - contextPath: MSGraphMail.FileAssessment.CreatedDateTime
      description: The datetime the file assessment was created.
      type: String
    - contextPath: MSGraphMail.FileAssessment.ContentType
      description: The content type.
      type: String
    - contextPath: MSGraphMail.FileAssessment.ExpectedAssessment
      description: The expected assessment.
      type: String
    - contextPath: MSGraphMail.FileAssessment.Category
      description: The category of the file assessment.
      type: String
    - contextPath: MSGraphMail.FileAssessment.Status
      description: The status of the file assessment.
      type: String
    - contextPath: MSGraphMail.FileAssessment.RequestSource
      description: The request source of the file assessment.
      type: String
    - contextPath: MSGraphMail.FileAssessment.FileName
      description: The file name of the file assessment.
      type: String
    - contextPath: MSGraphMail.FileAssessment.CreatedUserID
      description: The user ID that was created for the file assessment.
      type: String
    - contextPath: MSGraphMail.FileAssessment.CreatedUsername
      description: The user name that was created for the mail assessment.
      type: String
    - contextPath: MSGraphMail.FileAssessment.ResultType
      description: The result type of the file assessment.
      type: String
    - contextPath: MSGraphMail.FileAssessment.ResultMessage
      description: The result message of the file assessment.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 5]
```diff
@@ -1623,7 +1624,409 @@ script:
...
+      type: String
+  - arguments:
+    - description: expected assessment.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:1866]:
```suggestion
    - description: The expected assessment.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 6]
```diff
@@ -1623,7 +1624,409 @@ script:
...
+      isArray: false
+      name: expected_assessment
+      defaultValue: all
+      auto: PREDEFINED
+      predefined:
+      - 'block'
+      - 'unblock'
+      required: true
+    - description: category.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:1875]:
```suggestion
    - description: The category of the threat.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 7]
```diff
@@ -1623,7 +1624,409 @@ script:
...
+      isArray: false
+      name: category
+      defaultValue: all
+      auto: PREDEFINED
+      predefined:
+      - 'phishing'
+      - 'malware'
+      required: true
+    - description: url.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:1884]:
```suggestion
    - description: The URL of the threat.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 8]
```diff
@@ -1623,7 +1624,409 @@ script:
...
+      isArray: false
+      name: url
+      required: true
+    - description: request_id (generated in polling command).
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:1888]:
```suggestion
    - description: The request_id (generated in the polling command).
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 9]
```diff
@@ -1623,7 +1624,409 @@ script:
...
+      name: request_id
+      deprecated: true
+    description: Create url assessment request.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:1891]:
```suggestion
    description: Create the URL assessment request.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 10]
```diff
@@ -1623,7 +1624,409 @@ script:
...
+    - description: category.
+      isArray: false
+      name: category
+      defaultValue: all
+      auto: PREDEFINED
+      predefined:
+      - 'phishing'
+      - 'malware'
+      required: true
+    - description: file name.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:1806]:
```suggestion
    - description: The file name.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 11]
```diff
@@ -1623,7 +1624,409 @@ script:
...
+      description: 'Created user id.'
+      type: String
+    - contextPath: MSGraphMail.UrlAssessment.CreatedUsername
+      description: 'Created user name.'
+      type: String
+    - contextPath: MSGraphMail.UrlAssessment.ResultType
+      description: 'Result Type.'
+      type: String
+    - contextPath: MSGraphMail.UrlAssessment.ResultMessage
+      description: 'Result message.'
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:1936]:
```suggestion
      description: The mail assessment ID.
      type: String
    - contextPath: MSGraphMail.UrlAssessment.Url
      description: The URL of the URL assessment.
      type: string
    - contextPath: MSGraphMail.UrlAssessment.CreatedDateTime
      description: The datetime the URL assessment was created.
      type: String
    - contextPath: MSGraphMail.UrlAssessment.ContentType
      description: The content type of the URL assessment.
      type: String
    - contextPath: MSGraphMail.UrlAssessment.ExpectedAssessment
      description: The expected assessment of the URL assessment.
      type: String
    - contextPath: MSGraphMail.UrlAssessment.Category
      description: The category of the URL assessment.
      type: String
    - contextPath: MSGraphMail.UrlAssessment.Status
      description: The status of the URL assessment.
      type: String
    - contextPath: MSGraphMail.UrlAssessment.RequestSource
      description: The request source for the URL assessment.
      type: String
    - contextPath: MSGraphMail.UrlAssessment.RecipientEmail
      description: The recipient email for the URL assessment.
      type: String
    - contextPath: MSGraphMail.UrlAssessment.DestinationRoutingReason
      description: The destination routing reason for the URL assessment.
      type: String
    - contextPath: MSGraphMail.UrlAssessment.CreatedUserID
      description: The user ID created for the URL assessment.
      type: String
    - contextPath: MSGraphMail.UrlAssessment.CreatedUsername
      description: The user name created for the URL assessment.
      type: String
    - contextPath: MSGraphMail.UrlAssessment.ResultType
      description: The result type of the URL assessment.
      type: String
    - contextPath: MSGraphMail.UrlAssessment.ResultMessage
      description: The result message of the URL assessment.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 12]
```diff
@@ -1623,7 +1624,409 @@ script:
...
+      type: String
+  - arguments:
+    - description: request id.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:1939]:
```suggestion
    - description: The request ID.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 13]
```diff
@@ -1623,7 +1624,409 @@ script:
...
+      isArray: false
+      name: request_id
+      required: false
+    - description: filter, Example:category eq 'malware’ and ContentType eq 'file'. Available Fields are expectedAssessment,ContentType,status,requestSource.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:1943]:
```suggestion
    - description: The filter for the request. Example: category eq 'malware’ and ContentType eq 'file'. Available fields are expectedAssessment, ContentType, status, requestSource.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 14]
```diff
@@ -1623,7 +1624,409 @@ script:
...
+      isArray: false
+      name: filter
+      required: false
+    - description: order by.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:1947]:
```suggestion
    - description: The order by which to retrieve the results.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 15]
```diff
@@ -1623,7 +1624,409 @@ script:
...
+      predefined:
+      - 'id'
+      - 'createdDateTime'
+      - 'ContentType'
+      - 'expectedAssessment'
+      - 'category'
+      - 'status'
+      - 'requestSource'
+      required: false
+    - description: sort order.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:1960]:
```suggestion
    - description: The sort order.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 16]
```diff
@@ -1623,7 +1624,409 @@ script:
...
+      isArray: false
+      name: sort_order
+      auto: PREDEFINED
+      predefined:
+      - 'asc'
+      - 'desc'
+      required: false
+    - description: limit.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:1968]:
```suggestion
    - description: The maximum number of results to retrieve.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 17]
```diff
@@ -1623,7 +1624,409 @@ script:
...
+      isArray: false
+      name: limt
+      required: false
+    - description: next token.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:1972]:
```suggestion
    - description: The next token.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 18]
```diff
@@ -1623,7 +1624,409 @@ script:
...
+      isArray: false
+      name: next_token
+      required: false
+    description: Create url assessment request.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:1976]:
```suggestion
    description: Create URL assessment request.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 19]
```diff
@@ -1623,7 +1624,409 @@ script:
...
+      description: 'Created user id.'
+      type: String
+    - contextPath: MSGraphMail.AssessmentRequest.CreatedUsername
+      description: 'Created user name.'
+      type: String
+    - contextPath: MSGraphMail.AssessmentRequest.ResultType
+      description: 'Result Type.'
+      type: String
+    - contextPath: MSGraphMail.AssessmentRequest.ResultMessage
+      description: 'Result message.'
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:2027]:
```suggestion
    - contextPath: MsGraph.AssessmentRequestNextToken.next_token
      description: The next token for the assessment request.
      type: String
    - contextPath: MSGraphMail.AssessmentRequest.ID
      description: The assessment request ID.
      type: String
    - contextPath: MSGraphMail.AssessmentRequest.CreatedDateTime
      description: The datetime the assessment request was created.
      type: String
    - contextPath: MSGraphMail.AssessmentRequest.ContentType
      description: The content type of the assessment request.
      type: String
    - contextPath: MSGraphMail.AssessmentRequest.ExpectedAssessment
      description: The expected assessment of the assessment request.
      type: String
    - contextPath: MSGraphMail.AssessmentRequest.Category
      description: The category of the assessment request.
      type: String
    - contextPath: MSGraphMail.AssessmentRequest.Status
      description: The status of the assessment request.
      type: String
    - contextPath: MSGraphMail.AssessmentRequest.URL
      description: The URL of the assessment request.
      type: String
    - contextPath: MSGraphMail.AssessmentRequest.RequestSource
      description: The request source for the assessment request.
      type: String
    - contextPath: MSGraphMail.AssessmentRequest.RecipientEmail
      description: The email recipient for the assessment request.
      type: String
    - contextPath: MSGraphMail.AssessmentRequest.FileName
      description: The file name of the assessment request.
      type: String
    - contextPath: MSGraphMail.AssessmentRequest.DestinationRoutingReason
      description: The destination routing reason for the assessment request.
      type: String
    - contextPath: MSGraphMail.AssessmentRequest.CreatedUserID
      description: The user ID create for the assessment request.
      type: String
    - contextPath: MSGraphMail.AssessmentRequest.CreatedUsername
      description: The user name created for the assessment request.
      type: String
    - contextPath: MSGraphMail.AssessmentRequest.ResultType
      description: The result type of the assessment request.
      type: String
    - contextPath: MSGraphMail.AssessmentRequest.ResultMessage
      description: The result message of the assessment request.
```

File: Packs/MicrosoftGraphSecurity/ReleaseNotes/2_2_5.md [hunk 20]
```diff
@@ -0,0 +1,7 @@
+
+#### Integrations
+
+##### Microsoft Graph Security
+
+- Added support for Threat Assessment functionality to helps organizations to assess the threat received by any user in a tenant.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/ReleaseNotes/2_2_5.md:6]:
```suggestion
- Added support for threat assessment functionality to help organizations assess the threat received by any user in a tenant.
```

**Comment by ShirleyDenkberg** [comment]
@merit-maita After reviewing and implementing edits in the yml file, please regenerate the readme.
@thefrieddan1 @DeanArbel @ostolero Doc review completed.

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py [hunk 21]
```diff
@@ -250,6 +253,78 @@ def purge_ediscovery_data(self, case_id, search_id, purge_type, purge_areas):
         }
         return self.ms_client.http_request(method='POST', url_suffix=url, json_data=body, resp_type='response')
 
+    def create_mail_assessment_request(self, recipient_email, expected_assessment, category, user_id, message_id):
+        url = 'informationProtection/threatAssessmentRequests'
```
Comment (thefrieddan1) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py:257]: this variable is showing multiple times in method scope
Please extract it to a global const and change its name to url_suffix

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py [hunk 22]
```diff
@@ -1274,8 +1350,267 @@ def test_function(client: MsGraphClient, args, has_access_to_context=False):
                      f'Please check authentication related parameters. [{response.status_code}]')
 
 
+def get_message_user(client, message_user):
```
Comment (thefrieddan1) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py:1353]: I would like to see a unit test for this method

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py [hunk 23]
```diff
@@ -1274,8 +1350,267 @@ def test_function(client: MsGraphClient, args, has_access_to_context=False):
...
+    is_email = re.search(EMAIL_REGEX, message_user)
+    if is_email:
+        user_result = (client.get_user_id(message_user)).get('value')
```
Comment (thefrieddan1) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py:1356]: does al response from client.get_user_id contain value ? can it return with no value and throw exception ?

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py [hunk 24]
```diff
@@ -1274,8 +1350,267 @@ def test_function(client: MsGraphClient, args, has_access_to_context=False):
...
+        if not user_result:
+            raise DemistoException(f'{message_user} is not a valid user')
+        return user_result[0].get('id')
```
Comment (thefrieddan1) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py:1359]: is it possible for user_result not to be None and not to have id ? causing an exception

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py [hunk 25]
```diff
@@ -1274,8 +1350,267 @@ def test_function(client: MsGraphClient, args, has_access_to_context=False):
...
+def create_url_assessment_request_command(args, client: MsGraphClient) -> PollResult | CommandResults:
+
+    request_id = args.get('request_id')
+
+    if not request_id:
+        result = client.create_url_assessment_request(args.get('expected_assessment'),
+                                                      args.get('category'),
+                                                      args.get('url'))
+        request_id = result.get('id')
+        # request_id = "9203569d-313e-420e-fb3d-08dbd25dafbe"
```
Comment (thefrieddan1) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py:1552]: please remove

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py (hunk 23 above)
Comment (merit-maita) [reply Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py:1356]: if there's no user, the value is returned as an empty list

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py (hunk 24 above)
Comment (merit-maita) [reply Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py:1359]: if value is not empty, its a list of user's dict, which in each dict there's an id for sure for the looked up user

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py [hunk 26]
```diff
@@ -1274,8 +1346,272 @@ def test_function(client: MsGraphClient, args, has_access_to_context=False):
...
+    if not request_id:
+        message_user = get_message_user(client, args.get('message_user'))
+        result = client.create_mail_assessment_request(args.get('recipient_email'),
+                                                       args.get('expected_assessment'),
+                                                       args.get('category'),
+                                                       message_user,
+                                                       args.get('message_id'))
+        request_id = result.get('id')
+
+    # result = client.get_threat_assessment_request_status(request_id)
```
Comment (thefrieddan1) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py:1451]: please remove this commented out line

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py [hunk 27]
```diff
@@ -1274,8 +1346,272 @@ def test_function(client: MsGraphClient, args, has_access_to_context=False):
...
+    result = client.get_threat_assessment_request(request_id)
+    status = result.get('status')
+    demisto.debug(f"status is: {status}")
+    if status == 'completed' or result.get('results'):
+        # result = client.get_threat_assessment_request(request_id)
```
Comment (thefrieddan1) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py:1456]: remove

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py [hunk 28]
```diff
@@ -1274,8 +1346,272 @@ def test_function(client: MsGraphClient, args, has_access_to_context=False):
...
+
+    if not request_id:
+        content_data = get_content_data(args.get('entry_id'), args.get('content_data'))
+        result = client.create_email_file_assessment_request(args.get('recipient_email'),
+                                                             args.get('expected_assessment'),
+                                                             args.get('category'),
+                                                             content_data)
+        request_id = result.get('id')
+    demisto.debug(f"got request id: {request_id}")
+    # result = client.get_threat_assessment_request_status(request_id)
```
Comment (thefrieddan1) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py:1487]: remove comment

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py [hunk 29]
```diff
@@ -1274,8 +1346,268 @@ def test_function(client: MsGraphClient, args, has_access_to_context=False):
...
+        else:
+            raise ValueError("Argument must be string or bytes")
+        return base64.b64encode(base64.b64decode(string_bytes)) == string_bytes
+    except Exception:
+        return False
+
+
+def get_content_data(entry_id, content_data):   # pragma: no cover
+
+    if not (entry_id or content_data) or (entry_id and content_data):
```
[the remark this answers is not in the document]
Comment (merit-maita) [reply Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py:1384]: its fails with an str and none type

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py [hunk 30]
```diff
@@ -1274,8 +1346,261 @@ def test_function(client: MsGraphClient, args, has_access_to_context=False):
...
+        raise DemistoException('One of entry_id or content_data arguments has to be provided.')
```
Comment (thefrieddan1) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py:1385]: Just one entry_id or content_data arguments must to be provided.

**Review by thefrieddan1 — APPROVED** [review]

**Pull request merged**